otel-extensions/src/test/java/org/hypertrace/agent/otel/extensions/config/EnvironmentConfigTest.java

```diff
@@ -90,5 +100,11 @@ public void systemProperties() {
     Assertions.assertEquals(
         StringValue.newBuilder().setValue("/path/2/jar.jar").build(),
         agentConfig.getJavaagent().getFilterJarPaths(1));
+
+    // Assert that metrics endpoint is added correctly if set
+    System.setProperty(EnvironmentConfig.REPORTING_METRIC_ENDPOINT, "https://:-)");
```

Comment:
I used this here and again calling `applyPropertiesAndEnvVars` because in assertions above it is testing without this property set. Then here it is updating the systemProperties and then testing. I changed `http` to `https` to be sure it is taking the systemProperty set now. 

Same for assertion below this.